### PR TITLE
L2 assignment

### PR DESCRIPTION
**Pull request by gawhadebhavna**
Completed 4 services.

### CODE REVIEW

File: service/order.rest.xml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        <method type="post">
+            <service name="tutorial.order.OrderServices.create#OrderHeader"/>
+        </method>
+    </resource>
+    <resource name="orderPart" require-authentication="anonymous-all">
+        <method type="post">
+            <service name="tutorial.order.OrderServices.create#OrderPart"/>
+        </method>
+    </resource>
+    <resource name="orderDetail" require-authentication="anonymous-all">
```
Comment (gurveenbagga) [inline service/order.rest.xml:15]: We can use this in the resource "orders" by mentioning different method types.

File: service/order.rest.xml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        <method type="get">
```
Comment (gurveenbagga) [inline service/order.rest.xml:16]: Can you explore the use of orderId in resource?

File: service/order.rest.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+            <service name="tutorial.order.OrderServices.get#ReturnOrderInfo"/>
```
Comment (gurveenbagga) [inline service/order.rest.xml:17]: Can we rename the service to get#Order?

File: service/order.rest.xml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        </method>
+    </resource>
+    <resource name="AllOrders" require-authentication="anonymous-all">
+        <method type="get">
+            <service name="tutorial.order.OrderServices.get#ReturnAllOrders"/>
```
Comment (gurveenbagga) [inline service/order.rest.xml:22]: Can we rename the service to get#Orders?

File: service/order.rest.xml [hunk 5]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<resource xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="http://moqui.org/xsd/rest-api-2.1.xsd"
+          name="order" displayName="Order REST API" version="${moqui_version}">
+    <resource name="orders" displayName="For order details" require-authentication="anonymous-all">
+        <method type="post">
+            <service name="tutorial.order.OrderServices.create#OrderHeader"/>
```
Comment (gurveenbagga) [inline service/order.rest.xml:7]: Can we rename the service to create#OrderHeader to create#Order

File: service/order.rest.xml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        </method>
+    </resource>
+    <resource name="orderPart" require-authentication="anonymous-all">
+        <method type="post">
+            <service name="tutorial.order.OrderServices.create#OrderPart"/>
```
Comment (gurveenbagga) [inline service/order.rest.xml:12]: Can we rename the service to add#OrderItems

File: service/tutorial/order/OrderServices.xml [hunk 7]
```diff
@@ -0,0 +1,399 @@
...
+        </description>
+        <in-parameters>
+            <parameter name="orderId" required="true"/>
+            <parameter name="partName"/>
+            <parameter name="facilityId" required="true"/>
+            <parameter name="shipmentMethodEnumId"/>
+            <parameter name="customerPartyId" required="true"/>
+            <parameter name="item_details" type="List">
+                <parameter name="itemMap" type="Map">
+                    <parameter name="productId"/>
```
Comment (gurveenbagga) [inline service/tutorial/order/OrderServices.xml:189]: Can you check if we can add required for the item_details fields?

File: service/tutorial/order/OrderServices.xml [hunk 8]
```diff
@@ -0,0 +1,399 @@
...
+            </iterate>
+        </actions>
+    </service>
+    <!-- get one order detail here by orderId -->
+    <service verb="get" noun="ReturnOrderInfo">
+        <in-parameters>
+            <parameter name="orderId"/>
+        </in-parameters>
+        <out-parameters>
+            <parameter name="order" type="List">
```
Comment (gurveenbagga) [inline service/tutorial/order/OrderServices.xml:215]: order->orders

File: service/tutorial/order/OrderServices.xml [hunk 9]
```diff
@@ -0,0 +1,399 @@
...
+            <entity-find-one entity-name="mantle.order.OrderHeader" value-field="order_detail">
+                <field-map field-name="orderId"/>
+            </entity-find-one>
+            <set field="order" from="[]"/>
+            <!--2. fetch data from orderPart -->
+            <entity-find entity-name="mantle.order.OrderPart" list="order_parts">
+                <econdition field-name="orderId"/>
+            </entity-find>
+            <!--3.here setting customerPartyId for use in list customerDetails-->
+            <set field="customerPartyId" from="order_parts.customerPartyId"/>
```
Comment (gurveenbagga) [inline service/tutorial/order/OrderServices.xml:265]: Can you check to use .first  with list, In this case, we have same customerPartyId set in the orderParts, but we need to make sure we retrieve the right data from the list.

File: service/tutorial/order/OrderServices.xml [hunk 10]
```diff
@@ -0,0 +1,399 @@
...
+                            </parameter>
+                        </parameter>
+                    </parameter>
+                </parameter>
+            </parameter>
+        </out-parameters>
+        <actions>
+            <!-- Fetch data from orderHeader -->
+            <entity-find entity-name="mantle.order.OrderHeader" list="orderList">
+
```
Comment (gurveenbagga) [inline service/tutorial/order/OrderServices.xml:348]: Please remove extra spaces wherever not needed.

File: service/order.rest.xml (hunk 1 above)
Comment (gawhadebhavna) [reply service/order.rest.xml:15]: Okay, Ma'am.

File: service/order.rest.xml (hunk 2 above)
Comment (gawhadebhavna) [reply service/order.rest.xml:16]: Sure Ma'am, I will.

File: service/order.rest.xml (hunk 3 above)
Comment (gawhadebhavna) [reply service/order.rest.xml:17]: Okay, Ma'am.

File: service/order.rest.xml (hunk 5 above)
Comment (gawhadebhavna) [reply service/order.rest.xml:7]: Sure,Ma'am

File: service/tutorial/order/OrderServices.xml (hunk 7 above)
Comment (gawhadebhavna) [reply service/tutorial/order/OrderServices.xml:189]: Okay, Ma'am.

File: service/tutorial/order/OrderServices.xml (hunk 10 above)
Comment (gawhadebhavna) [reply service/tutorial/order/OrderServices.xml:348]: Okay, Ma'am.

**Comment by gurveenbagga** [comment]
Please create PR for your own repo.

**Comment by gawhadebhavna** [comment]
Closing this PR. Will push code on own repo.

**Pull request closed without merging**